### PR TITLE
Patch/position refs

### LINKED ISSUE #1075: Position: provided refs aren't usable
reported by @willsoto 

https://pluralsight.slack.com/archives/C70HPSJPP/p1595254691323600

### PR DESCRIPTION
**Pull request by danethurber**
resolves #1075

### CODE REVIEW

File: packages/util/src/clone-element-with-ref.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { cloneElement } from 'react'
+
+import { isFunction, isRef } from './index.js'
+
+export function cloneElementWithRef(element, outerRef, props) {
+  const { ref } = element
+
+  return cloneElement(element, {
+    ...props,
+    ref: node => {
```
Comment (jaketrent) [inline packages/util/src/clone-element-with-ref.js:10]: Would you please help me understand the need for the different cases here?

File: packages/util/src/clone-element-with-ref.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { cloneElement } from 'react'
+
+import { isFunction, isRef } from './index.js'
+
+export function cloneElementWithRef(element, outerRef, props) {
+  const { ref } = element
```
Comment (jaketrent) [inline packages/util/src/clone-element-with-ref.js:6]: Huh, I didn't know you could get the ref from react elements like this.  I'm surprised this is exposed.  :themoreyouknow:

File: packages/position/src/react/index.js [hunk 3]
```diff
@@ -88,7 +91,7 @@ const Position = React.forwardRef((props, forwardedRef) => {
 
   return (
     <>
-      {target ? child : React.cloneElement(child, { ref })}
```
Comment (jaketrent) [inline packages/position/src/react/index.js:91]: Apparently this didn't work to pass ref as a prop.  Did you learn why?  If so, would you mind explaining for my benefit, please?

Comment (danethurber) [reply packages/position/src/react/index.js:91]: this overwrites the ref with the new ref we're providing.

File: packages/util/src/clone-element-with-ref.js (hunk 1 above)
Comment (danethurber) [reply packages/util/src/clone-element-with-ref.js:10]: sure can. 

`if (isRef(outerRef)) outerRef.current = node` just checks that the new ref we're appending exists

`if (isFunction(ref)) ref(node)` supports the older ref style
`else if (isRef(ref)) ref.current = node` supports the newer refs from `createRef`

File: packages/position/src/react/index.js (hunk 3 above)
Comment (jaketrent) [reply packages/position/src/react/index.js:91]: ah

**Comment by EdwardIrby** [comment]
Just curious if position used context to provide inline styles to portaled element would we need to clone?

**Comment by danethurber** [comment]
@EdwardIrby context wouldn't help to much in this case i don't think. we could provide the styles through a render prop and expect the implementor to get add the styles that way. or we might be able to get away from cloning by adding extra divs around the child nodes

**Pull request merged**